tests/upgrades/test_discovery.py

```diff
@@ -0,0 +1,79 @@
+"""Test Discovery Plugin related Upgrade Scenario's
+
+:Requirement: UpgradedSatellite
+
+:CaseAutomation: Automated
+
+:CaseLevel: Acceptance
+
+:CaseComponent: DiscoveryImage
+
+:Team: Rocket
+
+:TestType: Functional
+
+:CaseImportance: High
+
+:Upstream: No
+"""
+import re
+
+from packaging import version
+import pytest
+
+
+class TestDiscoveryImage:
+    """Pre-upgrade and post-upgrade scenarios to test Foreman Discovery Image version.
+
+    Test Steps:
+        1. Before Satellite upgrade, Check the FDI version on the Discovered host
+        2. Upgrade satellite.
+        3. Check the FDI version on the Discovered host, after upgrade to make sure its same or greater.
+    """
+
+    @pytest.mark.pre_upgrade
+    def test_pre_upgrade_fdi_version(self, target_sat, save_test_data):
+        """Test FDI version before upgrade.
+
+        :id: preupgrade-8c94841c-6791-4af0-aa9c-e54c8d8b9a92
+
+        :steps:
+            1. Check installed version of FDI
+
+        :expectedresults: Version should be saved and checked post-upgrade
+        """
+        target_sat.register_to_cdn()
+        target_sat.execute(
+            'yum -y --disableplugin=foreman-protector install foreman-discovery-image'
```

Comment:
Why not do it the documented way, i.e. foreman-maintain?

Reply:
This is something that I got inspired from - https://github.com/SatelliteQE/robottelo/blob/35dca5a00df0e6a42a14e00cf46ff5fcc91c7b8b/pytest_fixtures/core/broker.py#L62C26-L62C26

Reply:
Yes @lhellebr I have updated it, Thanks for your input.